File: test/ModularPipelines.UnitTests/Attributes/LinuxOnlyTestAttribute.cs
namespace ModularPipelines.UnitTests.Attributes;

public class LinuxOnlyTestAttribute() : SkipAttribute("Linux only test")
{
    public override Task<bool> ShouldSkip(TestRegisteredContext context)
    {
        return Task.FromResult(!OperatingSystem.IsLinux());
    }
}

File: test/ModularPipelines.UnitTests.FSharp/Attributes/LinuxOnlyTestAttribute.fs
namespace ModularPipelines.UnitTests.FSharp.Attributes

open System
open System.Threading.Tasks
open TUnit.Core

type LinuxOnlyTestAttribute() =
    inherit SkipAttribute("Linux only test")
    override _.ShouldSkip(_: TestRegisteredContext) =
        Task.FromResult(not (OperatingSystem.IsLinux()))
